Scripts/RemediationScripts/Remediate-EnableHTTPSForAPIsInAPIManagementServices.ps1

```diff
@@ -0,0 +1,760 @@
+
+    if (-not (Test-Path -Path $backupFolderPath))
+    {
+        New-Item -ItemType Directory -Path $backupFolderPath | Out-Null
+    }
+ 
+    Write-Host $([Constants]::DoubleDashLine)
+    Write-Host "[Step 3 of 4] Backing up API Management Services details:"
+    
+    # Backing up API Management Services details.
+    $backupFile = "$($backupFolderPath)\APIManagementServicesWithoutHttpsEnabled.csv"
+
+    $apiManagementServicesWithoutHttpsEnabled| Export-CSV -Path $backupFile -NoTypeInformation
+
+    if (-not $DryRun)
+    {
+        Write-Host "API Management Services details have been backed up to $($backupFile)" -ForegroundColor $([Constants]::MessageType.Update)
+        Write-Host "HTTPS URL Scheme will be enabled on the API(s) of all API Management Services." -ForegroundColor $([Constants]::MessageType.Warning)
+
+        if (-not $Force)
+        {
+            Write-Host "Do you want to enable HTTPS URL Scheme on the API(s) of all API Management Services? " -ForegroundColor $([Constants]::MessageType.Warning) -NoNewline
+            $userInput = Read-Host -Prompt "(Y|N)"
+
+            if($userInput -ne "Y")
+            {
+                Write-Host "HTTPS URL Scheme will not be enabled for any API(s) of API Management Service. Exiting..." -ForegroundColor $([Constants]::MessageType.Update)
+                break
+            }
+        }
+        else
+        {
+            Write-Host "'Force' flag is provided. HTTPS URL Scheme will be enabled on the API(s) of all API Management Services without any further prompts." -ForegroundColor $([Constants]::MessageType.Warning)
+        }
+
+        Write-Host $([Constants]::DoubleDashLine)
+        Write-Host "[Step 4 of 4] Enabling HTTPS URL Scheme for API(s) in API Management Services..." -ForegroundColor $([Constants]::MessageType.Warning)
+
+        # To hold results from the remediation.
+        $apiManagementServicesRemediated = @()
+        $apiManagementServicesSkipped = @()
+
+        $apiManagementServicesWithoutHttpsEnabled| ForEach-Object {
+            $apiManagementService = $_
+            $resourceGroupName = $_.ResourceGroupName
+           
+            Write-Host "Enabling HTTPS URL Scheme for API(s) of API Management Service: Resource ID - $($_.ResourceId), Resource Group Name - $($_.ResourceGroupName), Resource Name - $($_.ResourceName)" -ForegroundColor $([Constants]::MessageType.Warning)
+            
+            try
+            {   #To hold name of API(s) which are remediated and skipped.        
+               $listOfApisRemediated = @()
+                $listOfApisSkipped = @()
+                
+                $apiMgmtContext = New-AzAPIManagementContext -ResourceGroupName $ResourceGroupName -ServiceName $_.ResourceName
```

Comment:
The variable in L383 is `$resourceGroupName` - starts with a lower case, unlike `$ResourceGroupName`.